[pip] Python3 fails to cross-compile wheels

Is there any python3 expert who can help to solve this problem?

pip seems to not work for cross compiling wheels for python3 anymore.
Since 1 Dec 2020 the github action of #4149 fails to cross compile the wheels.
The previous action succeeded one day before (except for ppc853x) and the only difference to the faulty build is one additional package in requirements.txt.

May be we need to nail pip to a version <20.3 as I got warnings about changing the dependency resolver behaviour in pip for another constellation (see second log below)

***Any suggestions would be appreciated.***

---

Error log from github action run for #4149 at 1 Dec 2020
part of the build for arch-x64-6.1:
```
2020-11-30T20:57:23.0680753Z ===>  Processing wheels of python3
2020-11-30T20:57:23.0709265Z ===>  Using existing requirements file
2020-11-30T20:57:23.0749309Z ===>  Force cross-compile
2020-11-30T20:57:23.3686141Z 
2020-11-30T20:57:23.3686755Z Usage:   
2020-11-30T20:57:23.3687380Z   pip wheel [options] <requirement specifier> ...
2020-11-30T20:57:23.3688613Z   pip wheel [options] -r <requirements file> ...
2020-11-30T20:57:23.3689381Z   pip wheel [options] [-e] <vcs project url> ...
2020-11-30T20:57:23.3690111Z   pip wheel [options] [-e] <local project path> ...
2020-11-30T20:57:23.3690701Z   pip wheel [options] <archive url/path> ...
2020-11-30T20:57:23.3691046Z 
2020-11-30T20:57:23.3691502Z no such option: -b
2020-11-30T20:57:23.4055829Z make: *** [../../mk/spksrc.wheel.mk:61: build_wheel_target] Error 2
2020-11-30T20:57:23.4057298Z make: Leaving directory '/github/workspace/spk/python3'
2020-11-30T20:57:23.4117644Z make: Entering directory '/github/workspace/spk/python3'
2020-11-30T20:57:23.4211474Z rm -fr work work-*
2020-11-30T20:57:24.4067079Z make: Leaving directory '/github/workspace/spk/python3'
```

---

It may be related to the follwing warning - seen on local builds running as local user (not root)
output from build of native/python3  
```
Collecting pip
  Using cached pip-20.3-py2.py3-none-any.whl (1.5 MB)
Collecting setuptools
  Using cached setuptools-50.3.2-py3-none-any.whl (785 kB)
Collecting wheel
  Using cached wheel-0.36.0-py2.py3-none-any.whl (34 kB)
Installing collected packages: pip, setuptools, wheel
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

homeassistant 0.118.2 requires bcrypt==3.1.7, which is not installed.
homeassistant 0.118.2 requires cryptography==3.2, which is not installed.
homeassistant 0.118.2 requires pyyaml==5.3.1, but you'll have pyyaml 5.3 which is incompatible.
homeassistant 0.118.2 requires ruamel.yaml==0.15.100, but you'll have ruamel-yaml 0.16.10 which is incompatible.
Successfully installed pip-20.3 setuptools-50.3.2 wheel-0.36.0
Collecting setuptools==44.1.0
  Using cached setuptools-44.1.0-py2.py3-none-any.whl (583 kB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 50.3.2
    Uninstalling setuptools-50.3.2:
      Successfully uninstalled setuptools-50.3.2
Successfully installed setuptools-44.1.0
Collecting cffi==1.14.4
  Downloading cffi-1.14.4-cp37-cp37m-manylinux1_x86_64.whl (402 kB)
     |████████████████████████████████| 402 kB 7.1 MB/s
Collecting pycparser
  Using cached pycparser-2.20-py2.py3-none-any.whl (112 kB)
Installing collected packages: pycparser, cffi
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pynacl 1.3.0 requires six, which is not installed.
Successfully installed cffi-1.14.4 pycparser-2.20
Collecting crossenv==1.0
  Downloading crossenv-1.0-py3-none-any.whl (21 kB)
Installing collected packages: cross
```

--- a/mk/spksrc.common.mk
+++ b/mk/spksrc.common.mk
@@ -17,7 +17,7 @@ RUN = cd $(WORK_DIR)/$(PKG_DIR) && env $(ENV)
 PIP ?= pip
 # Why ask for the same thing twice? Always cache downloads
 PIP_CACHE_OPT ?= --cache-dir $(PIP_DIR)
-PIP_WHEEL_ARGS = wheel --no-binary :all: $(PIP_CACHE_OPT) --no-deps --requirement $(WORK_DIR)/wheelhouse/requirements.txt --wheel-dir $(WORK_DIR)/wheelhouse -b $(WORK_DIR)/wheelbuild
+PIP_WHEEL_ARGS = wheel --no-binary :all: $(PIP_CACHE_OPT) --no-deps --requirement $(WORK_DIR)/wheelhouse/requirements.txt --wheel-dir $(WORK_DIR)/wheelhouse
 PIP_WHEEL = $(PIP) $(PIP_WHEEL_ARGS)
 
 # Available languages

--- a/mk/spksrc.python-wheel.mk
+++ b/mk/spksrc.python-wheel.mk
@@ -27,10 +27,10 @@ PYTHONPATH = $(PYTHON_LIB_NATIVE):$(INSTALL_DIR)$(INSTALL_PREFIX)/$(PYTHON_LIB_D
 build_python_wheel:
 ifeq ($(strip $(CROSSENV)),)
 # Python 2 way
-	@$(RUN) PYTHONPATH=$(PYTHONPATH) $(HOSTPYTHON) -c "import setuptools;__file__='setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" $(BUILD_ARGS) bdist_wheel -b $(WORK_DIR)/wheelbuild -d $(WORK_DIR)/wheelhouse
+	@$(RUN) PYTHONPATH=$(PYTHONPATH) $(HOSTPYTHON) -c "import setuptools;__file__='setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" $(BUILD_ARGS) bdist_wheel -d $(WORK_DIR)/wheelhouse
 else
 # Python 3 case: using crossenv helper
-	@. $(CROSSENV) && $(RUN) PYTHONPATH=$(PYTHONPATH) python -c "import setuptools;__file__='setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" $(BUILD_ARGS) bdist_wheel -b $(WORK_DIR)/wheelbuild -d $(WORK_DIR)/wheelhouse
+	@. $(CROSSENV) && $(RUN) PYTHONPATH=$(PYTHONPATH) python -c "import setuptools;__file__='setup.py';exec(compile(open(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" $(BUILD_ARGS) bdist_wheel -d $(WORK_DIR)/wheelhouse
 endif
 
 install_python_wheel: $(WHEEL_TARGET)

